### ISSUE TITLE
loss = NAN with revnet models on Cifar dataset

### ISSUE THREAD

**Issue by lexbee** [tbung/revnet#1]
Hey!

First of all, thank you for your hard work. I think reversible networks are very interesting and will open up a lot of new use-cases for ML. 

I noticed two things in using your code. 

First of all, I had to rename the revnet subdirectory to models because train_cifar.py imports models and then models.revnet. 

Secondly (and more importantly), when I run train_cifar.py --model revnet38 my loss quickly goes to NAN and the accuracy never surpasses 10%. Same thing for revnet110. The resnet architectures on the other hand all work perfectly fine with good accuracy. Any idea what's going on?

**Comment by tbung** [comment]
Hey!

Thanks for showing interest in my project! Unfortunately it is kind of a mess, that I should really clean up now that people are showing interest in it 😅 

Regarding your issues:

I am not at home until next week so I am not able to test anything right now. However you can try adding `--clip 0.25` when calling `train_cifar.py`. I found the gradients of the revnet tend to explode a bit and gradient norm clipping is what I used to obtain the results in the README.

I will add a task to the projects to add some kind of config storage that includes hyperparameters.

Hit me up if you know if it worked or not.

Happy holidays!

**Comment by lexbee** [comment]
Hey man,
Adding clipping worked beautifully! I really appreciate you getting back to me so quickly, especially during the holiday seasons. Happy holidays and all the best to you and your family!

**Comment by tbung** [comment]
Happy to hear that!
I will close this for now and add a note to the README.

I hope you had a wonderful holiday season!
